runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/registry/DispatchAgent.java

```diff
@@ -765,6 +776,14 @@ public void onExporterDetached(
         }
     }
 
+    @Override
+    public CatalogHandler supplyCatalog(
+        long catalogId)
+    {
+        CatalogRegistry catalog = configuration.resolveCatalog(catalogId);
+        return catalog != null ? catalog.handler() : null;
+    }
+
```

Comment:
Is this needed?

Note: we have a special case for some of the metrics related aspects to support `zilla metrics` command, which may not apply to catalogs.